toolchain/check/facet_type.cpp

```diff
@@ -654,87 +650,41 @@ auto ResolveFacetTypeRewriteConstraints(
 
     auto replace_witness_callbacks =
         SubstImplWitnessAccessCallbacks(&context, loc_id, &rewrite_values);
-    auto subst_inst_id =
+    auto subst_rhs_id =
         SubstInst(context, constraint.rhs_id, replace_witness_callbacks);
-    constraint.rhs_id = subst_inst_id;
+    constraint.rhs_id = rewrite_values.SetFullyRewritten(
+        context, loc_id, *lhs_access, constraint.lhs_id, subst_rhs_id);
+
     if (constraint.rhs_id == SemIR::ErrorInst::InstId) {
```

Comment:
Maybe we could skip updating `rhs_id` here, and instead update it in the loop below when we extract it from the rewrite values list?